okhttp/src/main/java/okhttp3/internal/http/Http1xStream.java

```diff
@@ -406,9 +397,8 @@ public FixedLengthSource(long length) throws IOException {
     @Override public void close() throws IOException {
       if (closed) return;
 
-      if (bytesRemaining != 0
-          && !Util.discard(this, DISCARD_STREAM_TIMEOUT_MILLIS, MILLISECONDS)) {
-        unexpectedEndOfInput();
+      if (bytesRemaining != 0 && !Util.discard(this, DISCARD_STREAM_TIMEOUT_MILLIS, MILLISECONDS)) {
+        endOfInput(false);
```

Comment:
Shouldn't this be
```java
endOfInput(Util.discard(this, ..., ...));
```
so that successful consumption of remaining bytes allows connection re-use?